syslog-ng-ctl/syslog-ng-ctl.c

```diff
@@ -347,6 +377,129 @@ static GOptionEntry no_options[] =
   { NULL, 0, 0, G_OPTION_ARG_NONE, NULL, NULL }
 };
 
+static void
+set_console_echo(gboolean new_state)
+{
+  struct termios t;
+
+  if (tcgetattr(STDIN_FILENO, &t))
+    {
+      fprintf(stderr, "syslog-ng-ctl: error while tcgetattr: %s\n", strerror(errno));
+      return;
+    }
+
+  if (new_state)
+    t.c_lflag |= ECHO;
+  else
+    t.c_lflag &= ~((tcflag_t) ECHO);
+
+  if (tcsetattr(STDIN_FILENO, TCSANOW, &t))
+    fprintf(stderr, "syslog-ng-ctl: error while tcsetattr: %s\n", strerror(errno));
+}
+
+static void
+get_password_from_stdin(gchar *buffer, gulong *length)
+{
+  printf("enter password:");
+  set_console_echo(FALSE);
+  if (-1 == getline(&buffer, length, stdin))
+    {
+      fprintf(stderr, "error while reading password from terminal: %s", strerror(errno));
```

Comment:
It would be nice to restore the echo.

Reply:
Due to the g_assert_not_reached() in the next line, I do not see the benefit.